### PR TITLE
Add backend-to-backend communication

### PR DESCRIPTION
**Pull request by Aaron2000119**
Part of https://github.com/Gamify-IT/issues/issues/304.

To test start the docker-compose file, go to the lecture pse and go to the first minigame you find when you go left and up. This should say crosswordpuzzle. Play it and see if the percentage in the overworld changes.

Please don't merge this issue before changing back the compose files.

### CODE REVIEW

File: src/main/java/de/unistuttgart/crosswordbackend/service/GameResultService.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+   * @param gameResult extern gameResultDTO
+   * @param userId Id of the user
+   * @param accessToken accessToken of the user
+   * @throws IllegalArgumentException if at least one of the arguments is null
+   */
+  public void submitGameResult(GameResultDTO gameResult, String userId, String accessToken) {
+    if (gameResult == null || userId == null || accessToken == null) {
+      throw new IllegalArgumentException("gameResultDTO or userId is null");
+    }
+    int score = 100 * gameResult.getCorrectTiles() / gameResult.getNumberOfTiles();
```
Comment (maexled) [inline src/main/java/de/unistuttgart/crosswordbackend/service/GameResultService.java:35]: If I see this correctly, the user has full control of the API call how many correct tiles, and how many tiles he can specify. Check preconditions, like correct tiles < number of tiles

**Comment by delvh** [comment]
`src/test/resources/get-results-response.json` seems to have been pushed accidentally.
Or did you want to test something there?

**Comment by cdingm** [comment]
image
image
when i try to play the crosswordpuzzle i see no crosswordpuzzle after pressing start

**Review by maexled — APPROVED** [review]
Do not forget to set the crosswordpuzzle images in docker-compose to main and remove this branch from docker image deploy workflow.

**Review by marlau97 — APPROVED** [review]

**Pull request merged**